### PR TITLE
Add blocks for managing sprite groups

### PR DESCRIPTION
**Pull request by balderdash**
Sprite groups are the sets of sprites that share an avatar, e.g. all the cats or all the dogs. Wandering, chasing, and fleeing are the same as the behaviors of pigs, storm troopers, and rebel pilots in the starwars levels. See Ryan's spec here: https://docs.google.com/document/d/1Zg-5Oy2aPdlQoHIRFSixqR51t9w3yePG2JNLuJnPV8o/edit

image

I'm reusing the implementation in `Item` by making `Sprite` extend `Item` instead of `Collidable` directly. That change was cleaner than expected, except that Sprites use x and y to represent their top left corner, while Items use it for the center. Other than that I just had to account for the fact that sprites come in varying sizes in the out-of-bounds/wall collision checks.

### CODE REVIEW

File: apps/i18n/studio/en_us.json [hunk 1]
```diff
@@ -52,6 +56,8 @@
   "catProcedures": "Functions",
   "catText": "Text",
   "catVariables": "Variables",
+  "caveboy": "cave boy",
+  "cavegirl": "cave girl (Jasmine)",
```
Comment (Hamms) [inline apps/i18n/studio/en_us.json]: I could see it being a bit confusing that we sometimes include the name and sometimes not.

**Comment by Hamms** [comment]
this looks fantastic! I'm shocked at how little extra code this adds.

LGTM

**Comment by balderdash** [comment]
Just remembered that parameterized strings are a thing. @Hamms could you take a quick look at my last commit?

File: apps/i18n/studio/en_us.json (hunk 1 above)
Comment (balderdash) [reply apps/i18n/studio/en_us.json]: I was blindly reproducing the way they appear in the "set sprite" block. Any idea why only some of those show the name?

Comment (Hamms) [reply apps/i18n/studio/en_us.json]: I know that only some of them currently have canonical names, I assume that's why. But it definitely makes less sense in a context where we expect to have multiple of the same sprite on the screen; I'd leave the names out of this implementation.

**Comment by Hamms** [comment]
oh that is much cleaner. LGTM

File: apps/i18n/studio/en_us.json (hunk 1 above)
Comment (balderdash) [reply apps/i18n/studio/en_us.json]: Makes sense, done.

**Pull request merged**